pennylane/ops/qubit/matrix_ops.py

```diff
@@ -439,6 +455,12 @@ def compute_decomposition(D, wires):
     def adjoint(self):
         return DiagonalQubitUnitary(qml.math.conj(self.parameters[0]), wires=self.wires)
 
+    def pow(self, n):
+        if isinstance(self.data[0], list):
+            return [DiagonalQubitUnitary([(x + 0.0j) ** n for x in self.data[0]], wires=self.wires)]
```

Comment:
Is this to cast it to complex? Why is this necessary?

Reply:
```pycon
>>> -1 ** 0.5
-1
>>> (-1+0j) ** 0.5
(6e-17 + 1j)
>>> np.array(-1) **0.5
tensor(nan, requires_grad=True)
>>> np.array(-1+0j)**0.5
tensor(0.+1.j, requires_grad=True)
```